src/sentry/features/manager.py

```diff
@@ -66,8 +68,37 @@ def has(self, name, *args, **kwargs):
 
         >>> FeatureManager.has('organizations:feature', organization, actor=request.user)
         """
+        return self.build_checker().has(name, *args, **kwargs)
+
+    def build_checker(self):
+        """
+        Set up for a delayed execution of ``has``.
+
+        Successive calls to the returned checker have better performance than
+        repeatedly calling ``FeatureManager.has``, because it retains the set
+        of ``feature.handler.FeatureHandler`` objects. An instance of the
+        checker should be kept only in a short-lived context, because any
+        dynamic changes to plugins' feature handlers are not reflected in its
+        behavior.
+        """
+        return FeatureChecker(self)
+
+
+class FeatureChecker(object):
+    def __init__(self, manager):
+        from sentry.plugins.base import plugins
```

Comment:
Nit: Can we import this up the top?

Reply:
I will if it doesn't cause a circular import.

Reply:
This one needs to stay here.